pkg/network/state.go

```diff
@@ -232,6 +242,54 @@ func (ns *networkState) GetDelta(
 	}
 }
 
+func (ns *networkState) saveTelemetry(telemetry map[ConnTelemetryType]int64) {
```

Comment:
Could you add a comment explaining how this works with `getTelemetryDelta` to have per-client non-monotonic telemetry? I understand how it works now, but I had to read the code multiple times.